mx.py

```diff
@@ -3445,13 +3445,17 @@ def pull(self, vcdir, rev=None, update=False, abortOnError=True):
         rc = self._fetch(vcdir, abortOnError)
         if rc == 0 and update:
             active_branch = self._active_branch(vcdir, abortOnError)
-            refspec = '{0}:{1}'.format(rev if rev else 'HEAD', active_branch)
-            cmd = ['git', 'pull', 'origin', refspec]
-            self._log_pull(vcdir, rev)
-            out = OutputCapture()
-            rc = self.run(cmd, nonZeroIsFatal=abortOnError, cwd=vcdir, out=out)
-            logvv(out.data)
-            return rc == 0
+            if rev:
+                return self.update(vcdir, rev=rev, mayPull=False, clean=True, abortOnError=abortOnError)
+            else:
+                # refspec = '{0}:{1}'.format(rev if rev else 'HEAD', active_branch)
+                refspec = 'HEAD:{0}'.format(active_branch)
+                cmd = ['git', 'pull', 'origin', refspec]
```

Comment:
@gilles-duboscq stated that it seems wrong to execute a `pull` after having executed `fetch` given that `pull` also does a `fetch`. I'd like to get his input of this change before merging it.